### ISSUE TITLE
Random seed intialization

### ISSUE THREAD

**Issue by dmoonat** [ourownstory/neural_prophet#147]
Is there a way for reproducibility of results?
I tried numpy random seed initialization but results are slightly different even with default parameters, n_lags=0.
Let me know if i'm missing something

**Comment by ourownstory** [comment]
Hi Deepak,
I assume you used this code?
```python
import numpy as np
np.random.seed(0)
```
This set's numpy's random number generator, however, our variability of results comes from SGD finding different optima on different runs (in PyTorch). 
The majority of the randomness comes from the random initialization of weights, different learning rates and different shuffling of the dataloader. 

You should be able to control the torch random number generator by adding this to your code:
```python
import torch
torch.manual_seed(0)
```
You can read more here:
https://pytorch.org/docs/stable/notes/randomness.html

Note: have an eye on the automatic learning rate. if it still varies, set it manually.

**Comment by dmoonat** [comment]
Hey, I tried torch seed also which works fine but i realized that i have to restart the jupyter kernel for every run to get the same results.
Thanks for your help!

**Comment by ourownstory** [comment]
reopening: will add helper function.

**Comment by ourownstory** [comment]
Documented at the end of Quickstart in chapter "Reproducibility"
https://ourownstory.github.io/neural_prophet/

**Comment by dmoonat** [comment]
Great, It will be helpful for the community!
